File: adventofcode/cpp/year2019/day15/day15_test.cc
#include "adventofcode/cpp/year2019/day15/day15.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day15, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day15/actual.in");
  std::string output = "232";
  adventofcode::answer_t a = day15::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day15, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day15/actual.in");
  std::string output = "320";
  adventofcode::answer_t a = day15::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: adventofcode/java/year2016/day15/Day15Test.java
package adventofcode.java.year2016.day15;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day15Test {
    @Test
    public void part1Example() throws IOException {
        try (var input = new FileReader("adventofcode/java/testdata/year2016/day15/example")) {
            var output = "5";
            var result = Day15.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day15/actual.in")) {
            var output = "148737";
            var result = Day15.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day15/actual.in")) {
             var output = "2353212";
             var result = Day15.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}
